core/src/main/java/hudson/model/Label.java

```diff
@@ -389,7 +389,7 @@ public int getTiedJobCount() {
                         for (Item i : parent.getItems()) {
                             if (i instanceof AbstractProject) {
                                 final AbstractProject project = (AbstractProject) i;
-                                if (this.equals(project.getAssignedLabel())) {
+                                if (this.name.equals(project.getAssignedLabelString())) {
```

Comment:
I do not think this short-circuit will work for derived `Label` classes. Derived `Label` classes are entitled to override `equals` so we cannot assume that a sub-class compares based on `name` equality only.